src/attributes.md

```diff
@@ -104,456 +105,45 @@ grammar:
-The attributes are only suggestions, and the compiler may choose to ignore it.
-
-Both attributes can be used on closures, [functions] and function prototypes,
-although they do not do anything on function prototypes. When applied to a
-function in a [trait], they apply only to that function when used as a default
-function for a trait implementation and not to all trait implementations.
-
-#### `inline` Attribute
-
-The *`inline` attribute* suggests to the compiler that it should place a copy of
-the attributed function in the caller, rather than generating code to call the
-function where it is defined.
+> _MetaListNameValueStr_:\
+> &nbsp;&nbsp; [IDENTIFIER] `(` ( _MetaNameValueStr_ (`,` _MetaNameValueStr_)* `,`<sup>?</sup> )<sup>?</sup> `)`
 
-> ***Note***: The compiler automatically inlines functions based on internal
-> heuristics. Incorrectly inlining functions can actually make the program
-> slower, so this attribute should be used with care.
-
-There are three ways of using the inline attribute:
-
-* `#[inline]` hints the compiler to perform an inline expansion.
-* `#[inline(always)]` asks the compiler to always perform an inline expansion.
-* `#[inline(never)]` asks the compiler to never perform an inline expansion.
-
-#### `cold` Attribute
-
-The *`cold` attribute* suggests to the compiler that the attributed function or
-closure is unlikely to be called.
-
-### `derive`
-
-The `derive` attribute allows certain traits to be automatically implemented
-for data structures. For example, the following will create an `impl` for the
-`PartialEq` and `Clone` traits for `Foo`, and the type parameter `T` will be given
-the `PartialEq` or `Clone` constraints for the appropriate `impl`:
+Some examples of meta items are:
 
-```rust
-#[derive(PartialEq, Clone)]
-struct Foo<T> {
-    a: i32,
-    b: T,
-}
-```
+Style | Example
+------|--------
+_MetaWord_ | `no_std`
+_MetaNameValueStr_ | `doc = "example"`
+_MetaListPaths_ | `allow(unused, clippy::inline_always)`
+_MetaListIdents_ | `macro_use(foo, bar)`
+_MetaListNameValueStr_ | `link(name = "CoreFoundation", kind = "framework")`
 
-The generated `impl` for `PartialEq` is equivalent to
+## Active and inert attributes
 
-```rust
-# struct Foo<T> { a: i32, b: T }
-impl<T: PartialEq> PartialEq for Foo<T> {
-    fn eq(&self, other: &Foo<T>) -> bool {
-        self.a == other.a && self.b == other.b
-    }
-
-    fn ne(&self, other: &Foo<T>) -> bool {
-        self.a != other.a || self.b != other.b
-    }
-}
-```
+An attribute is either active or inert. During attribute processing, *active
+attributes* remove themselves from the thing they are on while *inert attributes*
+stay on.
 
-You can implement `derive` for your own traits through [procedural macros].
+The [`cfg`] and [`cfg_attr`] attributes are active. The [`test`] attribute is
+inert when compiling for tests and active otherwise. [Attribute macros] are
+active. All other attributes are inert.
```

Comment:
`test` is always active, when not compiling for tests it transforms its input into nothing and disappears itself too.
Also, `derive` is an active attribute too.